File: models/robot.go
package models

// Robot struct
type Robot struct {
	ID               int64  `orm:"auto;pk;type(bigint);column(id)" json:"id"`
	NickName         string `orm:"unique;type(char);column(nickname)" json:"nickname"`
	Avatar           string `orm:"type(char);column(avatar)" json:"avatar"`
	Welcome          string `orm:"column(welcome)" json:"welcome"`                         // 欢迎语
	Understand       string `orm:"column(understand)" json:"understand"`                   // 不明白语句
	Artificial       string `orm:"column(artificial)" json:"artificial"`                   // 关键词转人工
	KeyWord          string `orm:"column(keyword)" json:"keyword"`                         // 知识库默认匹配词
	TimeoutText      string `orm:"column(timeout_text)" json:"timeout_text"`               // 超时提示语
	NoServices       string `orm:"column(no_services)" json:"no_services"`                 // 无人工在线提示语
	LoogTimeWaitText string `orm:"column(loog_time_wait_text)" json:"loog_time_wait_text"` // 长时间等待提示
	Switch           int    `orm:"default(0);column(switch)" json:"switch"`                // 是否开启
	System           int    `orm:"default(0);column(system)" json:"system"`                // 系统内置
	Platform         int64  `orm:"type(bigint);column(platform)" json:"platform"`          // 服务那个平台
	UpdateAt         int64  `orm:"type(bigint);column(update_at)" json:"update_at"`
	CreateAt         int64  `orm:"auto_now_add;type(int64);null;column(create_at)" json:"create_at"`
}

File: ui/kefu_flutter/lib/models/robot.dart
class Robot {
  int id;
  String nickname;
  String avatar;
  String welcome;
  String understand;
  String artificial;
  String keyword;
  String timeoutText;
  String noServices;
  String loogTimeWaitText;
  int isRun;
  int system;
  int platform;
  int updateAt;
  int createAt;

  Robot(
      {this.id,
      this.nickname,
      this.avatar,
      this.welcome,
      this.understand,
      this.artificial,
      this.keyword,
      this.timeoutText,
      this.noServices,
      this.loogTimeWaitText,
      this.isRun,
      this.system,
      this.platform,
      this.updateAt,
      this.createAt});

  Robot.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    nickname = json['nickname'];
    avatar = json['avatar'];
    welcome = json['welcome'];
    understand = json['understand'];
    artificial = json['artificial'];
    keyword = json['keyword'];
    timeoutText = json['timeout_text'];
    noServices = json['no_services'];
    loogTimeWaitText = json['loog_time_wait_text'];
    isRun = json['switch'];
    system = json['system'];
    platform = json['platform'];
    updateAt = json['update_at'];
    createAt = json['create_at'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['id'] = this.id;
    data['nickname'] = this.nickname;
    data['avatar'] = this.avatar;
    data['welcome'] = this.welcome;
    data['understand'] = this.understand;
    data['artificial'] = this.artificial;
    data['keyword'] = this.keyword;
    data['timeout_text'] = this.timeoutText;
    data['no_services'] = this.noServices;
    data['loog_time_wait_text'] = this.loogTimeWaitText;
    data['switch'] = this.isRun;
    data['system'] = this.system;
    data['platform'] = this.platform;
    data['update_at'] = this.updateAt;
    data['create_at'] = this.createAt;
    return data;
  }
}
